samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -23,12 +23,28 @@
 import java.util.List;
 import org.I0Itec.zkclient.IZkChildListener;
 import org.I0Itec.zkclient.IZkDataListener;
+import org.I0Itec.zkclient.exception.ZkBadVersionException;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.samza.SamzaException;
 import org.apache.zookeeper.data.Stat;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
+/**
+ * This class creates a barrier for version upgrade.
```

Comment:
A general comment on documentation:
Explaining the code verbatim makes the documentation hard to read. Instead, it will be cleaner to point out what the different roles are (leader and non-leader), what is sequence of events (leader prepares barrier and members create node under barrier), what is the criteria for completion (processor Id count and values match) and what kind of failures are possible (time out).